packages/lit-html/src/helpers.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * When `condition` is true, returns the result of calling `trueCase()`, else
+ * returns the result of calling `falseCase()` if `falseCase` is defined.
+ *
+ * This is a convenience wrapper around a ternary expression that makes it a
+ * little nicer to write an inline conditional without an else.
+ *
+ * @example
+ *
+ * ```ts
+ * render() {
+ *   return html`
+ *     ${when(this.user, () => html`User: ${this.user.username}, () => html`Sign In...`)}
+ *   `;
+ * }
+ * ```
+ */
+export const when = (
+  condition: boolean,
+  trueCase: () => unknown,
+  falseCase?: () => unknown
+): unknown => (condition ? trueCase() : falseCase?.());
```

Comment:
How about this to get stronger types?

```ts
function when<T>(condition: boolean, trueCase: () => T): T|undefined;
function when<T, F>(condition: boolean, trueCase: () => T, falseCase: () => F): T|F;
function when(
  condition: boolean,
  trueCase: () => unknown,
  falseCase?: () => unknown
) {
  return condition ? trueCase() : falseCase?.();
}
```

https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAdwBYFMwB4AqA+ACggQBMZYEAuRAIzjgBt0BDMAGkSgCcR0BhZgGd01AgEpEAXjyIcY6jgA+4EumAww6EgG4AUKEgUkaTLg4AxQsTBkj1OoxbtOPfkJGJxUmTg7BmDMICwqIS0ojm8rKK5noG0PDGGGAEuoiI1raJ9vRMrGxpLrzBHl7h4ADWYHDI7IX+gW7CAPyh3oiV1bW6EgDehVzoUCBcSJnkiYjNRU3oXtQNQe7NAHTiegC+urrWglCIAB5SKMkE3LwcZTIAjJdhMgBEwPQPYtpAA

(Also maybe there's a more compact way to do that)

Reply:
Or even this???

```ts
function when<T>(condition: true, trueCase: () => T): T;
function when<T>(condition: false, trueCase: () => T): undefined;
function when<T>(condition: boolean, trueCase: () => T): T|undefined;
function when<T, F>(condition: true, trueCase: () => T, falseCase: () => F): T;
function when<T, F>(condition: false, trueCase: () => T, falseCase: () => F): F;
function when<T, F>(condition: boolean, trueCase: () => T, falseCase: () => F): T|F;
function when(
  condition: boolean,
  trueCase: () => unknown,
  falseCase?: () => unknown
) {
  return condition ? trueCase() : falseCase?.();
}
```

Reply:
Oh nice. I didn't do this because it's unlikely to matter in template usages, but it's easy enough and nice to do anyway.

I got this as a refinement of option 2:
```ts
function when<T, F>(condition: true, trueCase: () => T, falseCase?: () => F): T;
function when<T, F = undefined>(condition: false, trueCase: () => T, falseCase?: () => F): F;
function when(
  condition: boolean,
  trueCase: () => unknown,
  falseCase?: () => unknown
) {
  return condition ? trueCase() : falseCase?.();
}
```

Reply:
This doesn't seem to work when the type is `boolean` -- only `true` or `false`:

https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAdwBYFMwB4AqAaRAMQD4AKCBAExlgQC5EoAnEdA51gYQEMBndBqQCUiALzFE+RMG4Abfj34B+QSPFEhDHAG4AUKEi0kaTLgKExicJXTAYYdJTIUw1Iwxny2jFukUDEYTEJKU8FPnQVQLUJQk0iPQNoeGMMMFJdREQXNxSGACM4OFl0bjA8TJ8uCNVgqzAAazA4ZHLKsL8IqKD1cCaWsF0RAG9KpnQoECYkHJoUxCUqzv4gjzlw5QA6YT0AX11dF14oRG4ARgYwEABbfPQmSxN0jm8eiTOhPSOT7gAmBmstnsjkeaVIHQIb0QHy+CGOpwAzJcbncmAAfQF2ByUUGYUgvU68RCFYqlcrROowg7fYkXRBXW73XHPXyQmLQtl1ABEwCKXM+hzhJ3y-0QxyY9gA5szwetXuyzpz1Dy+QKafkkfSUfc0eKpTKCXxiUUSmUle9zYgVXB+dogA